src/components/Heading/Heading.tsx

```diff
@@ -1,41 +1,61 @@
-import { SIZES } from "../../constants/sizes";
+import { DialogPositions } from "../../constants/sizes";
 import React, { useLayoutEffect } from "react";
-import PropTypes from "prop-types";
 import cx from "classnames";
 import Tooltip from "../../components/Tooltip/Tooltip";
 import useIsOverflowing from "../../hooks/useIsOverflowing";
 import useStyle from "../../hooks/useStyle";
 import useRefWithCallback from "../../hooks/useRefWithCallback";
-import TextWithHighlight from "../../components/TextWithHighlight/TextWithHighlight";
-import { TYPES } from "./HeadingConstants";
+import TextWithHighlight from "../TextWithHighlight/TextWithHighlight";
+import { HeadingSizes, HeadingTypes } from "./HeadingConstants";
+import VibeComponentProps from "../../types/VibeComponentProps";
 import "./Heading.scss";
 
-const Heading = ({
+export interface HeadingProps extends VibeComponentProps {
+  type?: HeadingTypes;
+  ariaLabel?: string;
+  value?: string;
+  ellipsis?: boolean;
+  ellipsisMaxLines?: number;
+  suggestEditOnHover?: boolean;
+  nonEllipsisTooltip?: string;
+  size?: typeof HeadingSizes;
+  highlightTerm?: string;
+  customColor?: string;
+  /** Custom font flag, use to enable new font family on H1 headers */
+  brandFont?: boolean;
+  style?: CSSStyleDeclaration;
+  tooltipPosition?: typeof DialogPositions;
+}
+
+const Heading: React.FC<HeadingProps> & {
+  sizes?: typeof HeadingSizes;
+  types?: typeof HeadingTypes;
+} = ({
   className,
-  value,
-  type,
-  size,
-  ariaLabel,
+  value = "",
+  type = HeadingTypes.h1,
+  size = HeadingSizes.LARGE,
+  ariaLabel = "",
   id,
   customColor,
-  ellipsis,
-  ellipsisMaxLines,
+  ellipsis = true,
+  ellipsisMaxLines = 1,
   style,
   tooltipPosition,
-  highlightTerm,
-  suggestEditOnHover,
-  brandFont,
-  nonEllipsisTooltip // tooltip to show when no overflow
+  highlightTerm = null,
+  suggestEditOnHover = false,
+  brandFont = false,
+  nonEllipsisTooltip = null // tooltip to show when no overflow
```

Comment:
let's move this comment to be on the interface declaration for be appeared on storybook

Reply:
Done